wdl-lint/src/v1/preamble_whitespace.rs

```diff
@@ -0,0 +1,192 @@
+//! A lint rule that checks for an incorrect preamble whitespace.
+
+use wdl_ast::experimental::v1::Visitor;
+use wdl_ast::experimental::AstNode;
+use wdl_ast::experimental::AstToken;
+use wdl_ast::experimental::Diagnostic;
+use wdl_ast::experimental::Diagnostics;
+use wdl_ast::experimental::Span;
+use wdl_ast::experimental::SyntaxElement;
+use wdl_ast::experimental::SyntaxKind;
+use wdl_ast::experimental::ToSpan;
+use wdl_ast::experimental::VersionStatement;
+use wdl_ast::experimental::VisitReason;
+use wdl_ast::experimental::Whitespace;
+
+use super::Rule;
+use crate::util::lines_with_offset;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the preamble whitespace rule.
+const ID: &str = "PreambleWhitespace";
+
+/// Creates an "unnecessary whitespace" diagnostic.
+fn unnecessary_whitespace(span: Span) -> Diagnostic {
+    Diagnostic::note("unnecessary whitespace in document preamble")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("remove the unnecessary whitespace")
+}
+
+/// Creates an "expected a blank line" diagnostic.
+fn expected_blank_line(span: Span) -> Diagnostic {
+    Diagnostic::note("expected a blank line before version statement")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a blank line between the last preamble comment and the version statement")
+}
+
+/// Detects incorrect whitespace in a document preamble.
+#[derive(Debug, Clone, Copy)]
+pub struct PreambleWhitespaceRule;
+
+impl Rule for PreambleWhitespaceRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that documents have correct whitespace in the preamble."
+    }
+
+    fn explanation(&self) -> &'static str {
```

Comment:
Your implementation of this rule is impressively thorough! You're catching lots of cases I hadn't thought about. I wonder if all the requirements for whitespace are being communicated via the below `explanation()`? I'm not sure 🤔 Maybe a simple addition like `No extraneous whitespace is allowed.` will cover all our bases?